test/tests/functional/pbs_sched_preempt_enforce_resumption.py

```diff
@@ -101,20 +106,343 @@ def test_suspended_job_ded_time_calendared(self):
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        j3 = Job(TEST_USER)
+        j3.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_q: 'expressq',
+                           ATTR_l + '.walltime': 8})
+        jid2 = self.server.submit(j3)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+
+        j3 = Job(TEST_USER)
+        j3.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_l + '.walltime': 10})
+        jid3 = self.server.submit(j3)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid3)
+
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid2, extend='x',
+                           offset=10, interval=2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid3)
+
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid3, extend='x',
+                           offset=15, interval=2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        p = '"express_queue, normal_jobs"'
+        a = {'preempt_prio': p}
+        self.server.manager(MGR_CMD_SET, SCHED, a, runas=ROOT_USER)
+
+        self.server.manager(MGR_CMD_UNSET, QUEUE, 'max_run_res_soft.ncpus',
+                            'workq')
+
+    def test_backfill_depth_starving_jobs(self):
+        """
+        This test confirms the working of Preemption along with starving jobs
+        and backfill_depth set
+        """
+        a = {ATTR_rescavail + '.ncpus': 8}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        a = {ATTR_qtype: 'Execution', ATTR_enable: 'True', ATTR_start: 'True'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "workq2")
+
+        self.server.manager(MGR_CMD_SET, SERVER, {ATTR_backfill_depth: 4})
+
+        a = {'help_starving_jobs': "True   ALL", 'max_starve': '00:00:10'}
+        self.scheduler.set_sched_config(a)
+
+        jids = []
+
+        j = Job(TEST_USER1)
+        j.set_attributes({ATTR_l + '.select': '1:ncpus=8',
+                          ATTR_q: 'workq2',
+                          ATTR_l + '.walltime': 300})
+        jid = self.server.submit(j)
+        jids.append(jid)
+
+        j = Job(TEST_USER1)
+        j.set_attributes({ATTR_l + '.select': '1:ncpus=6',
+                          ATTR_q: 'expressq',
+                          ATTR_l + '.walltime': 20,
+                          ATTR_l + '.preempt_targets': 'queue=workq2'})
+        jid = self.server.submit(j)
+        jids.append(jid)
+
+        walltimes = [300, 200, 300, 200, 300, 200, 300, 200, 190, 150, 10, 21]
+        ncpus = [8, 7, 6, 8, 7, 6, 1, 1, 1, 1, 1, 1]
+        states = ['S', 'R', 'Q', 'Q', 'Q', 'Q', 'Q',
+                  'Q', 'Q', 'Q', 'Q', 'Q', 'R', 'Q']
```

Comment:
You might consider using a list of dictionaries whose keys are 'walltime', 'ncpus', and 'state'.  This might be a little easier to keep track of when you are reading the code.